### PR TITLE
feat(frontend): add frontend metrics 

### PR DESCRIPTION
**Pull request by ZENOTME**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

support frontend metrics :
- query number of local execution (used to calculate qps of local execution mode)
- query latency of local execution 

this pr finish the task:
- add frontend metrics in frontend 
- add metrics_service in frontend 
- monitor metrics in frontend 

- modify RiseDev to generate frontend job in prometheus.yml
- modify RiseDev to support exporter_portal in frontend (similar to exporter_portal in compute node

- monitor frontend_metrics in grafana
image

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#5107

### CODE REVIEW

File: src/frontend/src/handler/query.rs [hunk 1]
```diff
@@ -55,6 +55,18 @@ pub async fn handle_query(
     };
     debug!("query_mode:{:?}", query_mode);
 
+    let timer = if query_mode == QueryMode::Local && stmt_type == StatementType::SELECT {
```
Comment (liurenjie1024) [inline src/frontend/src/handler/query.rs:58]: How about put this into `local_execute`?

File: grafana/risingwave-dashboard.py [hunk 2]
```diff
@@ -662,6 +667,23 @@ def section_batch_exchange(outer_panels):
...
+def frontend(outer_panels):
+    panels = outer_panels.sub_panel()
+    return [
+        outer_panels.row_collapsed("Frontend", [
+            panels.timeseries_query_per_sec("Query Per second in Loacl Execution Mode", [
+                panels.target(
+                    "rate(frontend_query_counter_local_execution[$__rate_interval])",""
+                ),
+            ]),
+            panels.timeseries_latency("Query Average Latency in Local Execution Mode", [
```
Comment (liurenjie1024) [inline grafana/risingwave-dashboard.py:679]: We should not use average here, rather generating quantiles such as p95, p50

File: src/frontend/src/monitor/stats.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+pub struct FrontendMetrics {
+    pub registry: Registry,
+    pub query_counter_local_execution: GenericCounter<AtomicU64>,
+    pub latency_local_execution: Histogram,
+}
+
+impl FrontendMetrics {
+    pub fn new(registry: Registry) -> Self {
+        let query_counter_local_execution = register_int_counter_with_registry!(
```
Comment (liurenjie1024) [inline src/frontend/src/monitor/stats.rs:28]: Will promethuese generate labeles such as host/port automatically?

Comment (ZENOTME) [reply src/frontend/src/monitor/stats.rs:28]: Yes

**Review by liurenjie1024 — APPROVED** [review]
LGTM

**Pull request merged**